tests/io/test_gempak.py

```diff
@@ -0,0 +1,210 @@
+    dimxr = gempak.IMXR.values
+    ddtar = gempak.DTAR.values
+
+    np.testing.assert_allclose(gpres, dpres, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gtemp, dtemp, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdwpt, ddwpt, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdrct, ddrct, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gsped, dsped, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(ghght, dhght, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gomeg, domeg, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gcwtr, dcwtr, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdtcp, ddtcp, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdtgp, ddtgp, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdtsw, ddtsw, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdtlw, ddtlw, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gcfrl, dcfrl, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gtkel, dtkel, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gimxr, dimxr, rtol=1e-10, atol=1e-2)
+    np.testing.assert_allclose(gdtar, ddtar, rtol=1e-10, atol=1e-2)
+
+
+@pytest.mark.parametrize('gem,gio,station', [
+    ('gem_sigw_hght_unmrg.csv', 'gem_sigw_hght_unmrg.snd', 'TOP'),
+    ('gem_sigw_pres_unmrg.csv', 'gem_sigw_pres_unmrg.snd', 'WAML')
+])
+def test_unmerged_sounding(gem, gio, station):
+    """Test loading an unmerged sounding.
+
+    PPBB and PPDD groups will be in height coordinates.
+    """
+    g = get_test_data(f'{gio}')
+    d = get_test_data(f'{gem}')
+
+    gso = GempakSounding(g).snxarray(station_id=f'{station}')[0]
+    gpres = gso.pres.values
+    gtemp = gso.temp.values.squeeze()
+    gdwpt = gso.dwpt.values.squeeze()
+    gdrct = gso.drct.values.squeeze()
+    gsped = gso.sped.values.squeeze()
+    ghght = gso.hght.values.squeeze()
+
+    gempak = pd.read_csv(d, na_values=-9999)
+    dpres = gempak.PRES.values
+    dtemp = gempak.TEMP.values
+    ddwpt = gempak.DWPT.values
+    ddrct = gempak.DRCT.values
+    dsped = gempak.SPED.values
+    dhght = gempak.HGHT.values
+
+    assert_allclose(gpres, dpres, rtol=1e-10, atol=1e-2)
+    assert_allclose(gtemp, dtemp, rtol=1e-10, atol=1e-2)
+    assert_allclose(gdwpt, ddwpt, rtol=1e-10, atol=1e-2)
+    assert_allclose(gdrct, ddrct, rtol=1e-10, atol=1e-2)
+    assert_allclose(gsped, dsped, rtol=1e-10, atol=1e-2)
+    assert_allclose(ghght, dhght, rtol=1e-10, atol=1e-1)
+
+
+def test_standard_surface():
+    """Test to read a standard surface file."""
+    def dtparse(string):
+        return datetime.strptime(string, '%y%m%d/%H%M')
+
+    skip = ['text']
+
+    g = get_test_data('gem_std.sfc')
+    d = get_test_data('gem_std.csv')
+
+    gsf = GempakSurface(g)
+    gstns = gsf.sfjson()
+
+    gempak = pd.read_csv(d, index_col=['STN', 'YYMMDD/HHMM'],
+                         parse_dates=['YYMMDD/HHMM'],
+                         date_parser=dtparse)
+    if not gempak.index.is_lexsorted():
```

Comment:
Looks like this block isn't needed?

Reply:
I was getting performance warnings from `pandas` for indexing past lexsort depth. Probably not necessary, but it did shut up a warning. I'll let you decide if it matters.

Reply:
Fair enough. Making pandas quieter is a good thing. 😉

Reply:
It looks like this call never returns `False` any more, so this block should probably be deleted.